mapper/thin_edge_json/src/json.rs

```diff
@@ -1,165 +1,250 @@
+                                                    name: k.into(),
+                                                }
+                                                .into(),
+                                            );
+                                        }
+                                        value => {
+                                            return Err(ThinEdgeJsonError::new_invalid_json_value(
+                                                k, value,
+                                            )
+                                            .into());
+                                        }
+                                    }
+                                }
+
+                                let () = visitor
+                                    .end_group()
+                                    .map_err(ThinEdgeJsonParserError::VisitorError)?;
                             }
 
                             _ => {
-                                return Err(ThinEdgeJsonError::new_invalid_json_value(k, v));
+                                return Err(
+                                    ThinEdgeJsonError::new_invalid_json_value(key, value).into()
+                                );
                             }
                         }
                     }
                 }
-                if measurements.is_empty() {
-                    Err(ThinEdgeJsonError::EmptyThinEdgeJsonRoot)
-                } else {
-                    Ok(ThinEdgeJson {
-                        timestamp: time,
-                        values: measurements,
-                    })
-                }
             }
-            _ => Err(ThinEdgeJsonError::new_invalid_json_root(&input)),
+            _ => return Err(ThinEdgeJsonError::new_invalid_json_root(&thin_edge_obj).into()),
         }
-    }
 
-    fn parse_from_rfc3339(timestamp: &str) -> Result<DateTime<FixedOffset>, ThinEdgeJsonError> {
-        let time = DateTime::parse_from_rfc3339(&timestamp).map_err(|err| {
-            ThinEdgeJsonError::InvalidTimestamp {
-                value: String::from(timestamp),
-                from: err,
-            }
-        })?;
-        Ok(time)
+        Ok(())
     }
 }
 
-impl SingleValueMeasurement {
-    fn new(name: impl Into<String>, value: f64) -> Result<Self, ThinEdgeJsonError> {
-        if value == 0.0 || value.is_normal() {
-            let single_value = SingleValueMeasurement {
-                name: name.into(),
-                value,
-            };
-            Ok(single_value)
-        } else {
-            Err(ThinEdgeJsonError::InvalidThinEdgeJsonNumber { name: name.into() })
+fn parse_from_rfc3339(timestamp: &str) -> Result<DateTime<FixedOffset>, ThinEdgeJsonError> {
+    let time = DateTime::parse_from_rfc3339(&timestamp).map_err(|err| {
+        ThinEdgeJsonError::InvalidTimestamp {
+            value: String::from(timestamp),
+            from: err,

```

Comment:
Nothing has changed in this function, why its showing as while function  has changed?

Reply:
It has moved due to addition of new functions.